src/spa/src/simple_parser/lexer.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <fstream>
+#include <vector>
+
+#include "simple_parser/token.h"
+
+namespace Simple {
+class Lexer {
+ public:
+  std::vector<Token*> tokens;
+  Lexer(std::istream& stream);
+};
+}  // namespace Simple
```

Comment:
I've seen the pattern of ending long brace blocks with a comment explaining which block they're closing. I really like it.

Reply:
actually clang-format generates this.